external-crates/move/crates/move-compiler/src/typing/expand.rs

```diff
@@ -368,6 +375,81 @@ fn inferred_numerical_value(
     }
 }
 
+fn inferred_string_value(
+    context: &mut Context,
+    _eloc: Loc,
+    value_loc: Loc,
+    value: Vec<u8>,
+    ty: &Type,
+) -> Option<T::Exp> {
+    use T::UnannotatedExp_ as E;
+    let diag_note =
+        || "String literals must be linked against the standard library to be constructed";
+
+    match &ty.value {
+        Type_::Apply(_, sp!(_, TypeName_::Builtin(sp!(_, bt))), args)
+            if bt.is_vector() && args.len() == 1 && core::is_u8_type(&args[0]) =>
+        {
+            Some(T::exp(
+                ty.clone(),
+                sp(value_loc, E::Value(sp(value_loc, Value_::Bytearray(value)))),
+            ))
+        }
+        Type_::Apply(_, sp!(_, name), args) if args.is_empty() => {
+            let possibles = context.outer.get_stdlib_string_info();
+            println!("ty: {ty:#?}");
+            println!("possibles: {possibles:#?}");
+
+            for (mut str_ty, str_ctor) in possibles {
```

Comment:
I'm a bit confused on how this works here. I would imagine that it would be something more like the following:
- Check using `.is(...)` which case we are in
- If it does not match, print the list of expected options

Reply:
And then we will need to decide what to do if we did not infer the type. I think we should either pick a default `std::string::String` seems fair? 🤷 or require an annotation? Thoughts?

Reply:
The current default is bytestrings, since there is no strong guarantee that `std` is loaded. We can change that, though.